numpy/core/tests/test_regression.py

```diff
@@ -1396,28 +1396,28 @@ def test_object_array_to_fixed_string(self):
 
     def test_unicode_to_string_cast(self):
         # Ticket #1240.
-        a = np.array([[u'abc', u'\u03a3'],
-                      [u'asdf', u'erw']],
+        a = np.array([['abc', '\u03a3'],
+                      ['asdf', 'erw']],
                      dtype='U')
         assert_raises(UnicodeEncodeError, np.array, a, 'S4')
 
     def test_unicode_to_string_cast_error(self):
         # gh-15790
-        a = np.array([u'\x80'] * 129, dtype='U3')
+        a = np.array(['\x80'] * 129, dtype='U3')
         assert_raises(UnicodeEncodeError, np.array, a, 'S')
         b = a.reshape(3, 43)[:-1, :-1]
         assert_raises(UnicodeEncodeError, np.array, b, 'S')
 
     def test_mixed_string_unicode_array_creation(self):
-        a = np.array(['1234', u'123'])
+        a = np.array(['1234', '123'])
```

Comment:
From the test name, this one actually looks like the first instance in the array was actually *intended* to be a Python2-style byte string. IIUC I think the test should be updated to include the `b` prefix for the array elements that are not marked with `u`.

Reply:
good catch, I've update this test to interleave string and byte objects